src/dev/formatWhitespace.m

```diff
@@ -0,0 +1,66 @@
+function formatWhitespace()
+    % FORMATWHITESPACE Formats whitespace in the active document
+    %
+    %   Rules:
+    %       - Smart-indent
+    %       - Indent content of all function with 4 spaces per level
+    %       - Remove whitespace in empty lines
+    %       - Preserve indentantion after line continuations, i.e. ...
+
+    import matlab.desktop.editor.*
+    try
+        d        = getActive();
+        oldLines = textToLines(d.Text);
+    catch
+        error('formatWhitespace:noEditorApi','Check that the Editor API is available.')
+    end
+
+    % Parse file into a tree
+    tree = mtree(d.Text);
```

Comment:
Yeah! It's called trial-and-error hahah (and reading the code...)

I used it in my https://github.com/okomarov/justify/blob/master/justify.m and you can check the use case.